### ISSUE TITLE
White screen, Web UI won't load in 0.3.26 win64

### ISSUE THREAD

**Issue by vt-idiot** [xbapps/xbvr#887]
Browser console log is complaining about MIME types?
```
app.aa21acea.js:1 
        
       Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/plain". Strict MIME type checking is enforced for module scripts per HTML spec.


chunk-vendors.a83622b9.js:1 
        
       Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/plain". Strict MIME type checking is enforced for module scripts per HTML spec.
```

Using an extension to overwrite the Content-Type response header allows XBVR to load like before:

![image](https://user-images.githubusercontent.com/81622808/180628074-538fafc4-447f-4a7a-a23e-41a9b9b5b46a.png)

Not sure which dependency/update bricked it, but I hope this helps you figure it out.

**Comment by crwxaj** [comment]
I'm afraid this is a case of "Works for me". I downloaded [xbvr_0.3.26_Windows_amd64.zip](https://github.com/xbapps/xbvr/releases/download/0.3.26/xbvr_0.3.26_Windows_amd64.zip), extracted and ran xbvr.exe and could access the Web UI in Chrome with no problem.
Can you give me the exact steps that lead to this issue?

**Comment by sheik124** [comment]
1. Downloaded it.
2. Dropped the .exe in place.
3. Ran it.
4. Attempted to access it using Chrome, Edge, or Firefox, from the same machine, or another machine on the network.

I do run XBVR from a RAM Disk, and have "APPDATA" pointed there by a .bat script that launches it, but that shouldn't be affecting this.

```
SETLOCAL
SET APPDATA=.\
xbvr.exe
ENDLOCAL
```

**Comment by vt-idiot** [comment]
1. Downloaded it.
2. Dropped the .exe in place.
3. Ran it.
4. Attempted to access it using Chrome, Edge, or Firefox, from the same machine, or another machine on the network.

I do run XBVR from a RAM Disk, and have "APPDATA" pointed there by a .bat script that launches it, but that shouldn't be affecting this.

```
SETLOCAL
SET APPDATA=.\
xbvr.exe
ENDLOCAL
```

**Comment by crwxaj** [comment]
I still can't reproduce it. Can you show me the request and response headers of the JS files in question?

**Comment by vt-idiot** [comment]
Because you're running it from a clean install? It works if I do that.

![image](https://user-images.githubusercontent.com/81622808/180646505-2582f079-f231-4bb4-814f-dcb0212b5d10.png)

It does not work if I try to keep my old configuration.
```
HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Length: 148169
Content-Type: text/plain; charset=utf-8
Vary: Origin
Date: Sun, 24 Jul 2022 12:15:59 GMT

GET /ui/js/app.aa21acea.js HTTP/1.1
Host: localhost:9999
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:102.0) Gecko/20100101 Firefox/102.0
Accept: */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate, br
Connection: keep-alive
Referer: http://localhost:9999/ui/
Sec-Fetch-Dest: script
Sec-Fetch-Mode: cors
Sec-Fetch-Site: same-origin
DNT: 1
Sec-GPC: 1
```

**Comment by theRealKLH** [comment]
FYI... I seem to have the issue only in HereSphere (Quest version), don't have DeoVR installed at the moment. All browsers in Quest2 and Windows work fine for me.

**Comment by vt-idiot** [comment]
> FYI... I seem to have the issue only in HereSphere (Quest version), don't have DeoVR installed at the moment. All browsers in Quest2 and Windows work fine for me.

Is that with a pre-populated library or a clean install? It loads fine from a clean install, not sure what exactly breaks it afterwards.

I have yet to try in HereSphere (Steam) and can't check in DeoVR since it crashes to desktop _and_ corrupts SteamVR's settings, breaking desktop views until I restart.

**Comment by vt-idiot** [comment]
This is bizarre, even the clean install returns those errors if I try to run XBVR from my server as I normally would.

I could only get it working properly with the clean run on a different Windows 10 machine. I dropped the main.db file from the server into %APPDATA%\xbvr\ to see if it works from this one and it worked fine.

Broken: Windows Server 2019, Version 1809, 17763.2803
Working: Microsoft Windows, Version 21H2, 19044.1826

**Comment by vt-idiot** [comment]
**It likely has nothing to do with Vue!**

@theRealKLH my pull request won't fix it, but there's no need to go hunting for all imports to add .vue because that won't fix it either...

# TL;DR It's Microsoft's fault.

And then it's Go's fault for overriding MIME types using whatever go/mime finds in `HKEY_CLASSES_ROOT`!!! Go _finally_ fixed it [at the end of May](https://go-review.googlesource.com/c/go/+/406894/) / [Github Commit](https://github.com/golang/go/commit/52f68efa45a34e60e8fc5a2ad5fc124a865ac2a4)

Can advise Windows users to apply a registry edit that will instantly fix this OR can change web server code (server.go?) to force the MIME type for .js OR update whatever part of Go is involved. Deleting the registry entry works, I didn't want to break any file associations, so I changed the Content Type to the only thing that should ever be used as the MIME type, `text/javascript`

From an admin command prompt:
```
REG ADD "HKEY_CLASSES_ROOT\.js" /v "Content Type" /t REG_SZ /d "text/javascript" /f
```
**Close XBVR completely and re-launch it.**

## XBVR now works fine.

It only checks the registry when you launch the program. I already used a batch file to launch XBVR, from before the -localstorage option got added, so I just added a line **that re-runs that registry edit before launching it**.

![galaxy goorbos](https://user-images.githubusercontent.com/81622808/180660764-351c83a2-7f1b-407a-b89c-eef25e798222.jpg)

<details>
  <summary>Moar</summary>
  
`https://github.com/golang/go/issues/32350`

The actual problem (in my case) starts with [Visual Studio](https://developercommunity.visualstudio.com/t/installer-mime-type-for-js-extension-is-set-to-tex/954263) changing the Content Type for .js in my registry, FFS. And apparently it _does it again_ _**whenever it updates**_. Jerks. I am not sure why the issue never appeared before, I haven't touched VS in ages, no updates, haven't launched it, nada.

The machine XBVR works fine on:
```
Microsoft Windows [Version 10.0.19044.1826]
(c) Microsoft Corporation. All rights reserved.

C:\>reg query HKCR\.js /v "Content Type"


ERROR: The system was unable to find the specified registry key or value.

C:\>
```

The busted machine:
```
Microsoft Windows [Version 10.0.17763.2803]
(c) 2018 Microsoft Corporation. All rights reserved.

C:\Users\Administrator>reg query HKCR\.js /v "Content Type"

HKEY_CLASSES_ROOT\.js
    Content Type    REG_SZ    text/plain


C:\Users\Administrator>
```

~~I can't into Go, but it seems like this is the fix, potentially somewhere in [/pkg/server/server.go](https://github.com/xbapps/xbvr/blob/master/pkg/server/server.go)~~
~~`https://github.com/golang/go/issues/32350#issuecomment-1128475902`~~

~~Abandonded PR to override this honestly idiotic way of deciding MIME types for files that should **never** have a different MIME type (CSS, HTML, JS): https://go-review.googlesource.com/c/go/+/191917/~~
</details>

**Comment by omdbapi** [comment]
@vt-idiot thanks for the regedit fix!

**Comment by theRealKLH** [comment]
ok. so what actually broke the functionality?

**Comment by vt-idiot** [comment]
> ok. so what actually broke the functionality?

It would've been something Go or Go dependency related? Either that, or something in the way XBVR runs its web server used to previously override MIME-Type for one reason or another, and now it doesn't. One of the dead ends I went down was the way Vue handles minification, it can force headers on files, not sure if any of the UI edits would've changed that.

As far as I can tell, Go should've been looking in the registry for Content Type associations and overriding MIME-Type with them _since the beginning_, I'm not sure why 0.3.26 was where the issue actually manifested. Go's finally fixed the issue upstream, not sure how or where that'd get pulled down to XBVR

> And then it's Go's fault for overriding MIME types using whatever go/mime finds in `HKEY_CLASSES_ROOT`!!! Go _finally_ fixed it [at the end of May](https://go-review.googlesource.com/c/go/+/406894/) / [Github Commit](https://github.com/golang/go/commit/52f68efa45a34e60e8fc5a2ad5fc124a865ac2a4)

**Comment by badman6666** [comment]
I'm still having the issue with Heresphere WebUI not showing. Shows fine the browsers for PC that runs xbvr, just doesnt work inside headset (Quest 2). Tried the registry thing, it did not change anything.
Happened after I updated to newest XBVR. I had to recreate library from zero (deleted all files and db) because XBVR did not want to start after update.

**Comment by vt-idiot** [comment]
> I'm still having the issue with Heresphere WebUI not showing. Shows fine the browsers for PC that runs xbvr, just doesnt work inside headset (Quest 2). Tried the registry thing with xbvr restart (on PC that runs xbvr), it did not change anything. Happened after I updated to newest XBVR. I had to recreate library from zero (deleted all files and db) because XBVR did not want to start after update.

That likely wasn't directly related to this issue. If the Go is overriding the .js MIME-Type via the registry, the regular Web UI does not load either, on any browser.

**Comment by vt-idiot** [comment]
Should be fixed by #869 and #919 

Otherwise this is still easily fixed by launching xbvr using a .bat file prepended with `REG ADD "HKEY_CLASSES_ROOT\.js" /v "Content Type" /t REG_SZ /d "text/javascript" /f`

_if you aren't running xbvr in a server environment where everything runs as administrator anyways, you might run into some issues trying to use a single .bat file to make the registry edit and then launch XBVR_
* cmd "REG" requires elevated permissions
* that means XBVR will launch with elevated permissions
